src/trio/_util.py

```diff
@@ -398,7 +398,10 @@ def raise_single_exception_from_group(
     # immediately bail out if there's any KI or SystemExit
     for e in eg.exceptions:
         if isinstance(e, (KeyboardInterrupt, SystemExit)):
-            raise type(e) from eg
+            new_exc = type(e)(e.args)
+            if isinstance(e, SystemExit):
+                new_exc.code = e.code  # code can't be set in SystemExit constructor
```

Comment:
I don't think this is right:

> The constructor accepts the same optional argument passed to sys.exit(). If the value is an integer, it specifies the system exit status (passed to C’s exit() function); if it is None, the exit status is zero; if it has another type (such as a string), the object’s value is printed and the exit status is one.

Reply:
Actually Python's behavior here is interesting. The comment still seems incorrect, but I agree with this extra work:

```pycon
>>> se = SystemExit("test")
>>> se.code
'test'
>>> se.code = 1
>>> se
SystemExit('test')
>>> se.args
('test',)
>>> se.code
1
```

(you probably knew this which is why you added this, and just wrote the comment wrong or something.)

Reply:
ok I figured why it was not working for me and fixed
systemexit is not supposed to get actual string error msg, it's just `SystemExit(4)`

Reply:
Well it supports it so Trio should probably do the right thing! (I do think it's a very silly use case though, and if it were more than a 5 line change then maybe we shouldn't support it)